[Term Entry] Swift Arrays: .reduce() (#1139)
Codecademy/docs: content/swift/concepts/arrays/terms/reduce/reduce.md

---
Title: '.reduce()'

> Description: 'Loops/Iterate over every item in a sequence, combines them into one value, and returns the combined result'

Reviewer: ```suggestion
Description: 'Loops/Iterates over every item in a sequence, combines them into one value, and returns the combined result.'
```

> Description: 'Loops/Iterates over every item in a sequence, combines them into one value, and returns the combined result.'

Reviewer 2: ```suggestion
Description: 'Loops or iterates over every item in a sequence, combines them into one value, and returns the combined result.'
```

Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`.reduce()`** method produces a single value from the elements of an entire sequence using the given closure.

Reviewer 2: I like the explanation that is mentioned in the description. Maybe we can add that explanation here

> It Takes 2 arguments

Reviewer: The note about arguments should be in the Syntax section after the ```pseudo block. Also, the first line should read "The `.reduce()` method takes two arguments:" also put a line between the introduction and the list itself.

```pseudo

> arrayName.reduce(_:_:) 

Reviewer 2: Instead of using unnamed variables, can we create a sea tic name placeholder for these parameters?

Author: Hey @yangc95 I didn't get this point. Should we make the syntax as arrayName.reduce(initialValue,closure)?

```

The `.reduce()` method takes two arguments:

- initial value - To store the initial value or result returned by the closure from each iteration.

> - closure - It takes 2 arguments, first one is the result from previous execution of the closure and the other one is next item in the collection.

Reviewer 2: Initial value and closure work well as the placeholder, make sure to follow Swift naming convention! Also, place the pseudo parameter names in backticks

> - initial value - To store the initial value or result returned by the closure from each iteration.
> - closure - It takes 2 arguments, first one is the result from previous execution of the closure and the other one is next item in the collection.

Reviewer 2: ```suggestion
- `initialValue` stores the initial value or result returned by the closure from each iteration.
- `closure` takes 2 arguments. The first one is the result from previous execution of the closure, and the second one is next item in the collection.
```

## Example

```swift
var values = [2,3,4,5]

let sum  = values.reduce(0, { x, y in x + y })
print(sum)
```

> In the example above, We are calculating the sum of the elements where `0` is the initial value followed by the closure, `x` holds the result of the previous execution and `y` is the next item in Array. This will output:

Reviewer 2: ```suggestion
In the example above, the sum of the elements where `0` is the initial value followed by the closure is calculated. `x` holds the result of the previous execution and `y` is the next item in the array. This will output:
```
